hack/ifreload/ifreload.c

```diff
@@ -0,0 +1,210 @@
+#include <netlink/route/qdisc.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <errno.h>
+#include <stdarg.h>
+#include <stdbool.h>
+#include <time.h>
+#include <unistd.h>
+
+
+static const char *LINK_NAME = "eth0";
+
+static const int BOUNCE_INTERVAL = 600;
+static const int HWM = 10240 * 90 / 100;
+static const int HWM_COUNT = 10;
+static const int INTERVAL = 60;
+
+static int sock;
+
+
+static void
+__logger(const char *format, va_list args) {
+	time_t now = time(NULL);
+
+	char buf[64];
+	strftime(buf, sizeof(buf), "%Y/%m/%d %H:%M:%S %Z", localtime(&now));
+	fprintf(stderr, "%s: ", buf);
+
+    vfprintf(stderr, format, args);
+}
+
+
+static void
+logger(const char *format, ...) {
+	va_list args;
+
+    va_start(args, format);
+	__logger(format, args);
+    va_end(args);
+
+	fputc('\n', stderr);
+}
+
+
+static void
+logger_error(const char *format, ...) {
+	va_list args;
+
+    va_start(args, format);
+	__logger(format, args);
+    va_end(args);
+
+	fprintf(stderr, ": %s\n", strerror(errno));
+}
+
+
+static int
+ifupdown(bool up) {
+	struct ifreq ifreq;
+	memset(&ifreq, 0, sizeof(ifreq));
+
+	strncpy(ifreq.ifr_name, LINK_NAME, IFNAMSIZ);
+
+	if (ioctl(sock, SIOCGIFFLAGS, &ifreq)) {
+		logger_error("ioctl SIOCGIFFLAGS");
+		return -1;
+	}
+
+	if (up) {
+		logger("bringing up interface %s", LINK_NAME);
+		ifreq.ifr_flags |= IFF_UP;
+
+	} else {
+		logger("bringing down interface %s", LINK_NAME);
+		ifreq.ifr_flags &= ~IFF_UP;
+	}
+
+	if (ioctl(sock, SIOCSIFFLAGS, &ifreq)) {
+		logger_error("ioctl SIOCSIFFLAGS");
+		return -1;
+	}
+
+	sleep(5);
+
+	logger("done");
+
+	return 0;
+}
+
+
+static void
+cb(struct nl_object *obj, void *arg) {
+	struct rtnl_qdisc *qdisc = nl_object_priv(obj);
+	struct rtnl_tc *tc = TC_CAST(qdisc);
+
+	*(uint64_t *)arg = rtnl_tc_get_stat(tc, RTNL_TC_QLEN);
+}
+
+
+int
+main(int argc, char **argv) {
+	if (argc > 1)
```

Comment:
```suggestion
	if (argc > 1) {
```
please the brackets.. :-O

Reply:
next it'll be gotos...